plugins/sonarqube-backend/api-report.md

```diff
@@ -0,0 +1,67 @@
+## API Report File for "@backstage/plugin-sonarqube-backend"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+import { Config } from '@backstage/config';
+import express from 'express';
+import { Logger } from 'winston';
+
+// @public
+export function createRouter(options: RouterOptions): Promise<express.Router>;
+
+// @public
+export class DefaultSonarqubeInfoProvider implements SonarqubeInfoProvider {
+  static fromConfig(config: Config): DefaultSonarqubeInfoProvider;
+  getBaseUrl(instanceName: string): string;
+  getFindings(
+    componentKey: string,
+    instanceName: string,
+  ): Promise<SonarqubeFindings | undefined>;
+}
+
+// @public
+export interface Measure {
+  metric: string;
+  value: string;
+}
+
+// @public
+export interface RouterOptions {
+  logger: Logger;
+  sonarqubeInfoProvider: SonarqubeInfoProvider;
+}
+
+// @public
+export class SonarqubeConfig {
+  constructor(instances: SonarqubeInstanceConfig[]);
+  static fromConfig(config: Config): SonarqubeConfig;
+  getInstanceConfig(sonarqubeName?: string): SonarqubeInstanceConfig;
+  // (undocumented)
+  readonly instances: SonarqubeInstanceConfig[];
+}
+
+// @public
+export interface SonarqubeFindings {
+  analysisDate: string;
+  measures: Measure[];
+}
+
+// @public
+export interface SonarqubeInfoProvider {
+  getBaseUrl(instanceName: string): string;
```

Comment:
I'd recommend wrapping up the arguments in an object so that you can make changes to the API easier in the future.

Something like `getBaseUrl({ instanceName }: { instanceName: string })` and then also returning an object rather than just a string. So the full signature would be `getBaseUrl({ instanceName }: { instanceName: string }): { baseUrl: string })`

I can see there's a few of these in this file, but I'd recommend moving them over.